internal/stage_file_search_3.go

```diff
@@ -0,0 +1,46 @@
+package internal
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/codecrafters-io/grep-tester/internal/test_cases"
+	"github.com/codecrafters-io/tester-utils/random"
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+func testMultiFileSearch(stageHarness *test_case_harness.TestCaseHarness) error {
+	RelocateSystemGrep(stageHarness)
+
+	file_name_1 := "fruits-" + randomFilePrefix() + ".txt"
+	file_name_2 := "vegetables-" + randomFilePrefix() + ".txt"
+	fruits := random.RandomElementsFromArray(FRUITS, random.RandomInt(2, 5))
```

Comment:
Based on your earlier comment that the setup stage can become long, let's shorten the randomness a bit then: 

```suggestion
	fruits := random.RandomElementsFromArray(FRUITS, random.RandomInt(2, 3))
```

Reply:
(Same applies for line below)